cmd/tuf/add_key.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/flynn/go-docopt"
+	"github.com/theupdateframework/go-tuf"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+func init() {
+	register("add-key", cmdAddKey, `
+usage: tuf add-key [--scheme=<scheme>] [--expires=<days>] [--public-value=<path>] <role>
+
+Adds a new signing key for the given role.
+
+The root metadata file will be staged
+with the addition of the key's ID to the role's list of key IDs.
+
+Options:
+  --public-value=<path>    The Path to the file containing value of the public key. If absent, will be read from stdin.
+  --expires=<days>    Set the metadata file to expire <days> days from now.
+  --scheme=<scheme>      Set the key scheme to use [default: ed25519].
+`)
+}
+
+func cmdAddKey(args *docopt.Args, repo *tuf.Repo) error {
+	role := args.String["<role>"]
+	var keyids []string
+
+	keyScheme := data.KeySchemeEd25519
+	switch t := args.String["--scheme"]; t {
+	case string(data.KeySchemeEd25519),
```

Comment:
I think we should not default to ed25519 here, I think it's better UX to abort with a nice error message like `key type %s is not supported` (where the variable `t` should be used during logging).